test/functional/ui/startup_spec.lua

```diff
@@ -0,0 +1,86 @@
+-- FIXME: ideally we'll eventually have tests for the entire startup sequence
+-- described at ":help initialization"
+-- For now, I'm just adding tests to ensure an init.lua gets loaded correctly
+
```

Comment:
there is `test/functional/core/startup_spec.lua`, I dont think this needs a separate `ui/startup_spec`.

Also look in `test/functional/core/startup_spec.lua` for how different startup options can be tested instead of refactoring `helpers.clear()`.